chapter03/Task04.py

```diff
@@ -7,17 +8,26 @@
       '"=" - Если я ответил верно')
 
 number = 50
+a = 0
+b = 101
+con = 0
 
 ans = str()
 while True:
-    ans = str(input('Это число' + str(number)))     #Внести правки по поводу диапозонов, проверки кода, дописать прогу
+    ans = input('Это число ' + str(number) + '?\n')
+    con += 1
     if ans == '=':
         break
     elif ans == '-':
-        number //= 2
+        b = number
+        number = b - (b - a) // 2
+        print(a, b)
     elif ans == '+':
-        number = number+(number//2)
+        a = number
+        number = a + (b - a) // 2
+        print(a, b)
     else:
```

Comment:
If you have "else" here, the user didn't input what you expect. I think it's a perfect place to output the user what you expect from him, e.g "please input + or -"